src/Controller/GameController.php

```diff
@@ -173,78 +177,87 @@ public function games(): JsonResponse
-
-        // Value are stored like "x-y".
-        $position = explode('-', $requestPosition);
+        if ($game->getTurnStatus() !== $this->gameService::ADD_MARBLE_STATUS) {
+            return new JsonResponse("This is not the add_marble phase", JsonResponse::HTTP_BAD_REQUEST);
+        }
 
-        // Using loop.index in twig make it start to 1 instead of 0.
-        // We need to remove 1 to each positions.
-        $position[0] -= 1;
-        $position[1] -= 1;
+        $position = $content["position"];
 
         $game = $this->gameService->addMarbleIfPositionIsValid($game, $position);
         $entityManager->flush();
-        $currentPlayerValue = $this->gameService->getPlayerValue($game, $playerHash);
+        $currentPlayerValue = $this->gameService->getCurrentPlayerValue($game);
+
         $this->notify(
             $publisher,
             $game->getId(),
             [
                 "status" => $this->gameService::ADD_MARBLE_STATUS,
-                "value" => ["position" => $requestPosition, "playerValue" => $currentPlayerValue]
+                "value" => ["position" => $position, "playerValue" => $currentPlayerValue],
             ]
         );
 
-        return $this->redirectToRoute('game', ["id" => $game->getId()]);
+        return JsonResponse::fromJsonString(
+            $this->serializer->serialize($game, 'json'),
+            JsonResponse::HTTP_OK,
+        );
     }
 
     /**
-     * @Route("/games/{id}/rotateQuarter", name="rotateQuarter")
+     * @Route("/games/{id}/rotateQuarter", name="rotateQuarter", methods={"POST"})
      * Add a marble to the board
      */
 
-    public function rotateQuarter(Request $request, string $id, PublisherInterface $publisher): Response
+    public function rotateQuarter(Request $request, string $id, PublisherInterface $publisher): JsonResponse
     {
-        $playerHash = $request->cookies->get($this::COOKIE_KEY);
-
+        $content = $request->toArray();
+        if (!$content || !$content["playerId"] || is_null($content["rotation"])) {
+            return new JsonResponse("playerId and rotation are required", JsonResponse::HTTP_BAD_REQUEST);
+        }
         $entityManager = $this->getDoctrine()->getManager();
         $game = $entityManager->getRepository(Game::class)->find($id);
 
-        if (!$playerHash || !$game) {
-            return $this->redirectToRoute('newGame');
+        if (!$game) {
+            return new JsonResponse("Game not found", JsonResponse::HTTP_NOT_FOUND);
         }
 
+        $playerId = $content["playerId"];
 
-        if ($game->getStatus() === $this->gameService::GAME_WAITING_OPPONENT) {
-            return $this->redirectToRoute('game', ["id" => $game->getId()]);
+        // $playerId is a string, we need to convert uuid to the same string.
+        if ($playerId !== $game->getCurrentPlayer()->getId()->toRfc4122()) {
+            return new JsonResponse("This is not your turn", JsonResponse::HTTP_BAD_REQUEST);
```

Comment:
As it is an authentication problem, it would be better to return in code `HTTP_UNAUTHORIZED` or `HTTP_FORBIDDEN`